caffe2/core/operator.h

```diff
@@ -420,8 +420,14 @@ class OperatorBase : public Observable<OperatorBase> {
 
 // OP_SINGLE_ARG provides a shorter initialization choice for initialization of
 // member variables for the class constructors.
+// This is a workaround for CUDA9.2 and GCC7
+#if defined(CUDART_VERSION) && CUDART_VERSION >= 9020 && __GNUC__ >= 7
+#define OP_SINGLE_ARG(type, name, variable, default)                           \
+  variable(this->template GetSingleArgument<type>(name, (default)))
```

Comment:
Is there any particular reason we couldn't just use this definition unconditionally? It seems like it should be valid for non CUDA9.2+GCC7 configs too.

Reply:
And indeed, in the rest of the patch you rewrite this unconditionally.